utils/composer-packages/src/ComposerPackageParser.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\ComposerPackages;
+
+use Composer\Semver\Semver;
+use Nette\Utils\Json;
+use Rector\Composer\ValueObject\PackageAndVersion;
+use Ssch\TYPO3Rector\ComposerPackages\Collection\ExtensionCollection;
+use Ssch\TYPO3Rector\ComposerPackages\ValueObject\ComposerPackage;
+use Ssch\TYPO3Rector\ComposerPackages\ValueObject\ExtensionVersion;
+use Ssch\TYPO3Rector\ComposerPackages\ValueObject\Typo3Version;
+
+final class ComposerPackageParser implements PackageParser
+{
+    public function parsePackage(string $content, ComposerPackage $composerPackage): ExtensionCollection
+    {
+        $json = Json::decode($content, Json::FORCE_ARRAY);
```

Comment:
Isn't it enough with the default PHP json functions? Why pull a external dependency here?

Reply:
Yes, should be enough. But then i have to implement the unhappy path myself. If the json is not valid, i have to catch this and throw an exception. The method is just convenient.